core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -110,6 +112,33 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     }
   }
 
+  /**
+    * Process a Mesos event and update internal state.
+    *
+    * @param event
+    * @return The events describing state changes as Mesos call intents
+    */
+  def handleMesosEvent(event: MesosEvent): SchedulerEvents = {
+    handleFrame { builder =>
+      builder.process { (specs, state, _) =>
+        mesosEventsLogic.processEvent(specs, state, cachedPendingLaunch.pendingLaunch)(event)
+      }
+    }
+  }
+
+  def handlePodRecordSnapshot(podRecords: Map[PodId, PodRecord]): SchedulerEvents = {
+    handleFrame { builder =>
+      builder.process { (specs, state, _) =>
+        if (state.podRecords.nonEmpty || specs.podSpecs.nonEmpty) {
```

Comment:
I don't understand this check: why shouldn't there be `podRecords` OR `podSpecs`?

Reply:
Since the snapshot is the first element that should/would be fed in to the scheduler graph, this is just a sanity check that everything so far is empty. Would you like to see it re-written in a different manner ?

Reply:
Also, see Tim's comment https://github.com/mesosphere/usi/pull/80#discussion_r275599366 I _think_ this code would be deleted eventually.

Reply:
What do you think about using `assert` instead?

Reply:
podRecords snapshot is staying